CoreFoundation/Base.subproj/SwiftRuntime/TargetConditionals.h

```diff
@@ -76,21 +77,31 @@
 #define TARGET_OS_LINUX        0
 #define TARGET_OS_WINDOWS      0
 #define TARGET_OS_BSD          0
+#define TARGET_OS_CYGWIN       0
 #elif __linux__
 #define TARGET_OS_DARWIN       0
 #define TARGET_OS_LINUX        1
 #define TARGET_OS_WINDOWS      0
 #define TARGET_OS_BSD          0
+#define TARGET_OS_CYGWIN       0
+#elif __CYGWIN__
+#define TARGET_OS_DARWIN       0
+#define TARGET_OS_LINUX        0
```

Comment:
So - how different is cygwin from linux? cygwin's idea is to provide a Linux-like environment for Windows.

I think we should have TARGET_OS_LINUX here set to 1 for cygwin, and also define DEPLOYMENT_TARGET_LINUX when building cygwin.

We can also define TARGET_OS_CYGWIN here (as you have done) and use that in the rare places where cygwin is truly different from Linux. That should dramatically simplify this patch, and also keep us from having to specify that deployment target everywhere (or forgetting to, and breaking the build for Cygwin).